detection_rules/rule_validators.py

```diff
@@ -98,6 +98,15 @@ def validate(self, data: 'QueryRuleData', meta: RuleMeta) -> None:
                 trailer = err_trailer
                 if "Unknown field" in message and beat_types:
                     trailer = f"\nTry adding event.module or event.dataset to specify beats module\n\n{trailer}"
+                elif "Field not recognized" in message:
+                    unrecognized_fields = []
+                    for field in self.unique_fields:
+                        if field in eql_schema.kql_schema:
+                            unrecognized_fields.append(eql_schema.kql_schema[field])
+                    field_types = ', '.join(unrecognized_fields)
+                    query = exc.source.strip()
+                    if unrecognized_fields:
+                        trailer = f"Query `{query}` uses an unsupported elasticsearch eql field_type `{field_types}`"
```

Comment:
we already have the ast parsed as a property, and we should consult the family [type](https://github.com/elastic/detection-rules/blob/46c2383e5b1247ac6c7e41aea4f130beaf7ec57c/kql/parser.py#L48-L83), so to simplify and for resiliency to any text field, lets create the following method (below ast)


```python
    def text_fields(self, eql_schema: ecs.KqlSchema2Eql) -> List[str]:
        """Return a list of fields of type text."""
        from kql.parser import elasticsearch_type_family

        return [f for f in self.unique_fields if elasticsearch_type_family(eql_schema.kql_schema.get(f)) == 'text']
```

and then change

```suggestion
                elif message.startswith('Field not recognized'):
                    text_fields = self.text_fields(eql_schema)
                    if text_fields:
                        fields_str = ', '.join(text_fields)
                        trailer = f"\neql does not support text fields: {fields_str}\n\n{trailer}"
```

Reply:
The `elasticsearch_type_family` is perfect!